jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -16,203 +16,214 @@
+
+        if (filterBy.equals(CaseRolesAssignmentFilterBy.ASSIGNED.getLabel())) {
+            view.setRolesAssignmentList(
+                    allUnfilteredElements.stream()
+                            .filter(caseRoleAssignmentSummary -> hasAssignment(caseRoleAssignmentSummary))
+                            .collect(toList()));
+        } else if (filterBy.equals(CaseRolesAssignmentFilterBy.UNASSIGNED.getLabel())) {
+            view.setRolesAssignmentList(
+                    allUnfilteredElements.stream()
+                            .filter(caseRoleAssignmentSummary -> !hasAssignment(caseRoleAssignmentSummary))
+                            .collect(toList()));
+        } else {
+            view.setRolesAssignmentList(allUnfilteredElements);
         }
+    }
 
-        cis.getRoleAssignments().forEach(
-                crs -> {
-                    crs.getUsers().forEach(user -> {
-                        if (crs.getName().equals("owner")) {
-                            view.addUser(user, crs.getName());
-                        } else {
-                            view.addUser(user, crs.getName(), new CaseRoleAction() {
-
-                                @Override
-                                public String label() {
-                                    return translationService.format(REMOVE);
-                                }
-
-                                @Override
-                                public void execute() {
-                                    removeUserFromRole(user, crs.getName());
-                                }
-                            });
-                        }
-                    });
-                    crs.getGroups().forEach(group -> view.addGroup(group, crs.getName(), new CaseRoleAction() {
-
-                        @Override
-                        public String label() {
-                            return translationService.format(REMOVE);
-                        }
-
-                        @Override
-                        public void execute() {
-                            removeGroupFromRole(group, crs.getName());
-                        }
-                    }));
-                }
-        );
-    }
-
-    protected void addAssignment(final String userName,final String groupName, final String roleName) {
-        if(!isNullOrEmpty(userName) && !isNullOrEmpty(groupName)){
-            caseService.call(
-                    (Void) -> findCaseInstance()
-            ).assignGroupAndUserToRole(serverTemplateId, containerId, caseId, roleName, userName,groupName);
-        }else {
-            if (!isNullOrEmpty(userName)) {
-                addUserToRole(userName, roleName);
-            }
-            if(!isNullOrEmpty(groupName)){
-                addGroupToRole(groupName,roleName);
-            }
+    public void editAction(CaseRoleAssignmentSummary caseRoleAssignmentSummary) {
+
+        editRoleAssignmentView.setValue(CaseRoleAssignmentSummary.builder()
```

Comment:
here the role assignment is clone to allow the cancel action and restore the old values